ogre2/src/Ogre2DepthCamera.cc

```diff
@@ -583,15 +681,70 @@ void Ogre2DepthCamera::CreateDepthTexture()
       Ogre::CompositorPassSceneDef *passScene =
           static_cast<Ogre::CompositorPassSceneDef *>(
           colorTargetDef->addPass(Ogre::PASS_SCENE));
-      passScene->mVisibilityMask = IGN_VISIBILITY_ALL
-          & ~(IGN_VISIBILITY_GUI | IGN_VISIBILITY_SELECTABLE);
+      passScene->mVisibilityMask = IGN_VISIBILITY_ALL;
+
       // todo(anyone) Fix shadows. The shadow compositor node gets rebuilt
       // when the number of shadow-casting light changes so we end up with
       // invalid shadow node here. See Ogre2Scene::PreRender function on how
       // it destroys and triggers a compositor rebuild in OgreCamera when
       // the number of shadow-casting light changes
       // passScene->mShadowNode = "PbsMaterialsShadowNode";
     }
+
+    Ogre::CompositorTargetDef *depthTargetDef =
+        baseNodeDef->addTargetPass("depthTexture");
+    depthTargetDef->setNumPasses(2);
+    {
+      // clear pass
+      Ogre::CompositorPassClearDef *passClear =
+          static_cast<Ogre::CompositorPassClearDef *>(
+          depthTargetDef->addPass(Ogre::PASS_CLEAR));
+      passClear->mColourValue = Ogre::ColourValue::Black;
+      // scene pass
+      Ogre::CompositorPassSceneDef *passScene =
+          static_cast<Ogre::CompositorPassSceneDef *>(
+          depthTargetDef->addPass(Ogre::PASS_SCENE));
+      // depth texute does not contain particles
+      passScene->mVisibilityMask = IGN_VISIBILITY_ALL
+          & ~Ogre2ParticleEmitter::kParticleVisibilityFlags;
+    }
+
+    Ogre::CompositorTargetDef *particleTargetDef =
+        baseNodeDef->addTargetPass("particleTexture");
+    particleTargetDef->setNumPasses(2);
+    {
+      // clear pass
+      Ogre::CompositorPassClearDef *passClear =
+          static_cast<Ogre::CompositorPassClearDef *>(
+          particleTargetDef->addPass(Ogre::PASS_CLEAR));
+      passClear->mColourValue = Ogre::ColourValue(0, 0, 0, 0);
```

Comment:
Can we use `Ogre::ColourValue::Black`? Or does that not give us the alpha value we want?

Reply:
we can use black here as the texture format is a single channel grayscale that does not have alpha. 963bc43